### ISSUE TITLE
Diagnostic info - device list not expanding

### ISSUE THREAD

**Issue by judetf** [uvjim/linksys_velop#239]
[config_entry-linksys_velop-27e88a74ddd423caa49d0a3bc468f646.json.txt](https://github.com/uvjim/linksys_velop/files/9270430/config_entry-linksys_velop-27e88a74ddd423caa49d0a3bc468f646.json.txt)

**Comment by uvjim** [comment]
Hello.  Could you try this code for Card 2 please?

<details>
  <summary>Card 2 YAML</summary>

```yaml
type: custom:auto-entities
card:
  type: vertical-stack
card_param: cards
filter:
  include:
    - entity_id: /binary_sensor.(velop_)*mesh_wan_status/
      options:
        type: entities
        card_mod:
          style:
            .: |
              #states { padding-top: 0px; }
            fold-entity-row:
              $:
                template-entity-row:
                  $: |
                    state-badge { display: none; }
                    state-badge + div { margin-left: 8px !important; }
                    .info.pointer { font-weight: 500; }
                    .state { margin-right: 10px; }
        entities:
          - type: custom:button-card
            entity: this.entity_id
            show_name: false
            icon: hass:web
            tap_action:
              action: none
            custom_fields:
              attr_dns_servers: '[[[ return (entity != undefined) ? entity.attributes.dns : ""]]]'
              attr_public_ip: '[[[ return (entity != undefined) ? entity.attributes.ip : "" ]]]'
              attr_speedtest_latest: |
                [[[
                  let entity_speedtest_id = 'sensor.' + this._config.entity.split('.')[1].split('_').slice(0, -2).join('_') + '_speedtest_latest'
                  let entity_speedtest = states[entity_speedtest_id]
                  if (entity_speedtest != undefined && entity_speedtest.state != 'unknown') {
                    let d = new Date(entity_speedtest.state)
                    return "As at: " + d.toLocaleString()
                  } else {
                    return "No Speedtest results"
                  }
                ]]]
              attr_speedtest_details: |
                [[[
                  let round2 = (num) => Math.round(num * 100) / 100
                  let spacing_internal = 5
                  let spacing_external = 30
                  let icon_size = 22
                  let entity_speedtest_id = 'sensor.' + this._config.entity.split('.')[1].split('_').slice(0, -2).join('_') + '_speedtest_latest'
                  let entity_speedtest = states[entity_speedtest_id]
                  if (entity_speedtest != undefined && entity_speedtest.state != 'unknown') {
                    let latency = entity_speedtest.attributes.latency
                    let download_bandwidth = round2(entity_speedtest.attributes.download_bandwidth / 1000)
                    let upload_bandwidth = round2(entity_speedtest.attributes.upload_bandwidth / 1000)

                    return `<span style="margin-right: ${spacing_external}px;">
                              <ha-icon icon="hass:swap-horizontal" style="width: ${icon_size}px;"></ha-icon>
                              <span>${latency}ms</span>
                            </span>
                            <span style="margin-right: ${spacing_external}px;">
                              <ha-icon icon="hass:cloud-download-outline" style="width: ${icon_size}px;"></ha-icon>
                              <span>${download_bandwidth} Mbps</span>
                            </span>
                            <span>
                              <ha-icon icon="hass:cloud-upload-outline" style="width: ${icon_size}px;"></ha-icon>
                              <span>${upload_bandwidth} Mbps</span>
                            </span>
                            `
                  }
                ]]]
            state:
              - value: 'on'
                color: darkcyan
              - value: 'off'
                color: darkred
            styles:
              card:
                - box-shadow: none
                - padding: 16px 8px
              grid:
                - grid-template-areas: >-
                    "attr_dns_servers . attr_public_ip" "i i i"
                    "attr_speedtest_details attr_speedtest_details
                    attr_speedtest_details" "attr_speedtest_latest
                    attr_speedtest_latest attr_speedtest_latest"
                - grid-template-rows: 5% 1fr 15% 5%
                - grid-template-columns: 1fr min-content 1fr
              custom_fields:
                attr_dns_servers:
                  - justify-self: self-start
                attr_public_ip:
                  - justify-self: self-end
            extra_styles: >
              div[id^="attr_"] { font-size: smaller; color:
              var(--disabled-text-color);

              }

              div[id^="attr_speedtest_"] { margin-top: 10px; }

              #attr_public_ip::before { content: 'Public IP: ' }

              #attr_dns_servers::before { content: 'DNS: ' }
          - type: conditional
            conditions:
              - entity: binary_sensor.velop_mesh_speedtest_status
                state: 'on'
            row:
              type: section
          - type: conditional
            conditions:
              - entity: binary_sensor.mesh_speedtest_status
                state: 'on'
            row:
              type: section
          - type: custom:template-entity-row
            card_mod:
              style: >
                state-badge { display: none; } state-badge + div.info {
                margin-left: 8px !important; margin-right: 8px; text-align:
                center; }
            condition: >-
              {{ is_state('binary_sensor.velop_mesh_speedtest_status', 'on') or
              is_state('binary_sensor.mesh_speedtest_status', 'on')}}
            name: >-
              {% set entity_status = "this.entity_id" | replace("wan_status",
              "speedtest_status") %}  {{ state_attr(entity_status, 'status') }}
            tap_action:
              action: none
          - type: section
          - type: custom:auto-entities
            card:
              type: horizontal-stack
            card_param: cards
            sort:
              method: friendly_name
            filter:
              include:
                - entity_id: /^(button|switch)\.(velop_)*mesh_/
                  options:
                    type: custom:button-card
                    hold_action:
                      action: more-info
                    tap_action:
                      action: >-
                        [[[ return (entity.entity_id.startsWith("button")) ?
                        "call-service" : "toggle" ]]]
                      service: >-
                        [[[ return (entity.entity_id.startsWith("button")) ?
                        "button.press" : undefined ]]]
                      service_data:
                        entity_id: entity
                    name: |-
                      [[[
                        let friendly_name = entity.attributes.friendly_name.replace(/Velop|Mesh|:/gi, "").trim()
                        let idx = friendly_name.lastIndexOf(" ");
                        let ret = friendly_name.substring(0, idx) + "<br />" + friendly_name.substring(idx + 1)
                        return ret
                      ]]]
                    styles:
                      card:
                        - box-shadow: none
                        - margin-bottom: 3px
                      icon:
                        - animation: |-
                            [[[
                              let ret
                              let speedtest_entity = states["binary_sensor.velop_mesh_speedtest_status"] || states["binary_sensor.mesh_speedtest_status"]
                              if (entity.entity_id.match(/button\.(velop_)*mesh_start_speedtest/) && speedtest_entity.state == "on") {
                                ret = "rotating 2s linear infinite"
                              }
                              return ret
                            ]]]
                        - color: |-
                            [[[
                              let ret
                              let col_on = "darkcyan"
                              let col_off = "var(--primary-text-color)"
                              ret = (entity.state == "on") ? col_on : col_off
                              if (entity.entity_id.match(/button\.(velop_)*mesh_start_speedtest/)) {
                                let speedtest_entity = states["binary_sensor.velop_mesh_speedtest_status"] || states["binary_sensor.mesh_speedtest_status"]
                                ret = (speedtest_entity.state == "on") ? col_on : col_off
                              }
                              return ret
                            ]]]
                      name:
                        - font-size: smaller
                        - color: |-
                            [[[
                              let ret
                              let col_on = "darkcyan"
                              let col_off = "var(--primary-text-color)"
                              ret = (entity.state == "on") ? col_on : col_off
                              if (entity.entity_id.match(/button\.(velop_)*mesh_start_speedtest/)) {
                                let speedtest_entity = states["binary_sensor.velop_mesh_speedtest_status"] || states["binary_sensor.mesh_speedtest_status"]
                                ret = (speedtest_entity.state == "on") ? col_on : col_off
                              }
                              return ret
                            ]]]
          - type: section
          - type: custom:fold-entity-row
            padding: 0
            head:
              type: custom:template-entity-row
              entity: this.entity_id
              name: Online Devices
              state: >-
                {% set devices_entity = "sensor." ~
                "this.entity_id".split(".")[1] | replace("wan_status",
                "online_devices") %} {{ states(devices_entity) }}
              tap_action:
                action: fire-dom-event
                fold_row: true
              card_mod:
                style: |
                  .info.pointer { font-weight: 500; }
                  .state { margin-right: 10px; }
            entities:
              - type: custom:hui-element
                card_type: markdown
                card_mod:
                  style:
                    .: |
                      ha-card { border-radius: 0px; box-shadow: none; }
                      ha-markdown { padding: 16px 0px 0px !important; }
                    ha-markdown$: >
                      table { width: 100%; border-collapse: separate;
                      border-spacing: 0px; }

                      tbody tr:nth-child(2n+1) { background-color:
                      var(--table-row-background-color); }

                      thead tr th, tbody tr td { padding: 4px 10px; }
                content: >
                  {% set devices_entity = "sensor." ~
                  "this.entity_id".split(".")[1] | replace("wan_status",
                  "online_devices") %} {% set devices =
                  state_attr(devices_entity, "devices") %} | # | Name | IP |
                  Type |

                  |:---:|:---|---:|:---:| {% if devices %}
                    {%- for device in devices -%}
                      {%- set device_name = device.name if "name" in device else '<ha-icon icon="mdi:minus"></ha-icon>' -%}
                      {%- set device_ip = device.ip if "ip" in device else '<ha-icon icon="mdi:minus"></ha-icon>' -%}
                      {%- set connection_type = device.connection | lower if "connection" in device else "" -%}
                      {%- set guest_network = device.guest_network if "guest_network" in device else "" -%}
                      {%- if connection_type == "wired" -%}
                        {%- set connection_icon = "ethernet" -%}
                      {% elif connection_type == "wireless" -%}
                        {%- set connection_icon = "wifi" -%}
                      {% elif connection_type == "unknown" -%}
                        {%- set connection_icon = "help" -%}
                      {% else -%}
                        {%- set connection_icon = "minus" -%}
                      {%- endif %}
                  {{ "| {} | {}{} | {} | {} |".format(loop.index, device_name,
                  '&nbsp;<ha-icon icon="mdi:account-multiple"></ha-icon>' if
                  guest_network else '', device_ip, '<ha-icon icon="mdi:' ~
                  connection_icon ~ '"></ha-icon>') }}
                    {%- endfor %}
                  {% endif %}
          - type: custom:fold-entity-row
            padding: 0
            head:
              type: custom:template-entity-row
              entity: this.entity_id
              name: Offline Devices
              state: >-
                {% set devices_entity = "sensor." ~
                "this.entity_id".split(".")[1] | replace("wan_status",
                "offline_devices") %} {{ states(devices_entity) }}
              tap_action:
                action: fire-dom-event
                fold_row: true
              card_mod:
                style: |
                  .info.pointer { font-weight: 500; }
                  .state { margin-right: 10px; }
            entities:
              - type: custom:hui-element
                card_type: markdown
                card_mod:
                  style:
                    .: |
                      ha-card { border-radius: 0px; box-shadow: none; }
                      ha-markdown { padding: 16px 0px 0px !important; }
                    ha-markdown$: >
                      table { width: 100%; border-collapse: separate;
                      border-spacing: 0px; }

                      tbody tr:nth-child(2n+1) { background-color:
                      var(--table-row-background-color); }

                      thead tr th, tbody tr td { padding: 4px 10px; }
                content: >
                  {% set devices_entity = "sensor." ~
                  "this.entity_id".split(".")[1] | replace("wan_status",
                  "offline_devices") %} {% set devices =
                  state_attr(devices_entity, 'devices') %} | # | Name |
                  |:---:|:---| {% for device in devices %} {{ "| {} | {}
                  |".format(loop.index, device.name) }} {% endfor %}
          - type: custom:auto-entities
            card:
              type: custom:fold-entity-row
              head:
                type: section
                label: Storage
              padding: 0
            show_empty: false
            filter:
              include:
                - entity_id: /sensor.(velop_)*mesh_available_storage/
                  options:
                    type: custom:hui-element
                    card_type: markdown
                    card_mod:
                      style:
                        .: |
                          ha-card { border-radius: 0px; box-shadow: none; }
                          ha-markdown { padding: 16px 0px 0px !important; }
                        ha-markdown$: >
                          table { width: 100%; border-collapse: separate;
                          border-spacing: 0px; }

                          tbody tr:nth-child(2n+1) { background-color:
                          var(--table-row-background-color); }

                          thead tr th, tbody tr td { padding: 4px 10px; }
                    content: >
                      {% set storage_entity = "sensor." ~
                      "this.entity_id".split(".")[1] | replace("wan_status",
                      "available_storage") %} {% set partitions =
                      state_attr(storage_entity, 'partitions') %} | Host | Label
                      | %age used

                      |:---:|:---:|:---:|

                      {% for partition in partitions %} {{ "| {} | {} | {}
                      |".format(partition.ip, partition.label,
                      partition.used_percent) }}

                      {% endfor %}
```
</details>

**Comment by judetf** [comment]
That fixed it! Thank you so much!

**Comment by uvjim** [comment]
No problem. Could you do me a favour could you copy the contents of the entity in developer tools (as text, not a screenshot)? The one you took a screenshot of yesterday. I'd like to see if I can tally it back to see why some attributes are missing.

**Comment by judetf** [comment]
Sure. This is what's in there now, but I don't know if this has what you
need or not. Do you need me to reload the older card 2 code to reproduce
the error? Happy to do so if that helps.

Here's what the console log is at the moment:

button-card.js?hacstag=146194325342:425   BUTTON-CARD   Version 3.4.2
card-mod.js?hacstag=190927524315:5 CARD-MOD 3.1.5 IS INSTALLED
card-mod.js?hacstag=190927524315:5 You may not be getting optimal
performance out of card-mod.
See
https://github.com/thomasloven/lovelace-card-mod#performance-improvements
fold-entity-row.js?hacstag=150781994220:78 FOLD-ENTITY-ROW 2.2.0 IS
INSTALLED
hui-element.js?hacstag=249942054180547:1 HUI-ELEMENT 1.1.2 IS INSTALLED
template-entity-row.js?hacstag=231674882130:51 TEMPLATE-ENTITY-ROW 1.3.0 IS
INSTALLED
auto-entities.js?hacstag=1677445841110:172 AUTO-ENTITIES 1.11.0 IS INSTALLED
config-template-card.js?hacstag=172177543136:329 The main 'lit-element'
module entrypoint is deprecated. Please update your imports to use the
'lit' package: 'lit' and 'lit/decorators.ts' or import from
'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path
for more information.
(anonymous) @ config-template-card.js?hacstag=172177543136:329
config-template-card.js?hacstag=172177543136:735   CONFIG-TEMPLATE-CARD
 Version 1.3.6
state-switch.js?hacstag=158756598195:139 STATE-SWITCH 1.9.5 IS INSTALLED
hui-markdown-card.ts:77 Uncaught (in promise) Object
value @ hui-markdown-card.ts:77
hui-markdown-card.ts:77 Uncaught (in promise) Object
value @ hui-markdown-card.ts:77
/images/lovelace_light.png:1          Failed to load resource: the server
responded with a status of 404 (Not Found)
/lovelace-thermostats/0?edit=1:1 Autofocus processing was blocked because a
document already has a focused element.
core.f2e094cd.js:1 Uncaught (in promise) Object
ha-logbook.ts:42 card undefined Error: No card type configured
    at u (ha-logbook.ts:42:1)
    at d (ha-logbook.ts:42:1)
    at v (ha-logbook.ts:42:1)
    at c.value (hui-stack-card.ts:16:1)
    at hui-light-card.ts:35:1
    at Array.map (<anonymous>)
    at c.value (hui-light-card.ts:35:1)
    at Mt.update_card (auto-entities.js?hacstag=1677445841110:172:4764)
    at Mt.update_all (auto-entities.js?hacstag=1677445841110:172:3017)
d @ ha-logbook.ts:42

**Comment by uvjim** [comment]
Sorry, I meant from the Home Assistant Developer Tools.

![Screenshot_2022-08-05-14-04-58-58_c3a231c25ed346e59462e84656a70e50](https://user-images.githubusercontent.com/8431065/183251881-007f1e13-b9c3-4c3f-b842-ccc3349104d1.jpg)

Could you copy all the state attribute text and paste here please?

**Comment by judetf** [comment]
Ah, sorry, here it is:
options:
  - 020OPS-M-54MRQ
  - 07573b14-d717-407c-9761-440da07b9ab9
  - 0e09bb49fc2974e8
  - 0e09bb49fc2974e8
  - 10-238-1-148
  - 10-238-1-166
  - 10-238-1-32
  - 10.238.1.117 - Sonos Sub
  - 10.238.1.13 - Sonos Play:5
  - 10.238.1.168 - Sonos Play:5
  - 10.238.1.76 - Sonos Play:5
  - 13b1c8bd-c6ff-49f6-8728-ff501791e71e
  - 176d356ab41f5696769ba6295efa8e9e
  - 2760acd5f5bbc8da0791608d6206e720
  - 3570e8816f41d16e3fad588a613932ed
  - 482f942eb9cfaf511a9122a16238d4ba
  - 5c5926d056120989692cfe9b62745125
  - 5ddaa5cd6349bd4eded7482b62018853
  - 6a594c67-e5e4-4147-8083-51f2f84a60bf
  - 6f17ae9e-36a0-44fc-b1b9-2e17aaf34fcf
  - 6f1fd49d-7118-402b-8aea-d6960935afd5
  - 7449f3ca-9bc2-4ec7-bf71-c4ddbcca9e83
  - 7fcea139e58a202f9244b26e61b90b85
  - 84a445b3f9faa7af81d7ebd161fd2392
  - 9a2d2f90782108a0fb83ca8f61a93059
  - 9ae6a32d-0e2a-4201-b24c-2177cf6c9e6a
  - AmazonDevice
  - AmazonDevice
  - Android
  - Android-3
  - Apple-Watch
  - BRN30055C934CCE
  - BRN30055C934CCE
  - BRN30055C934CCE
  - BRN30055C934CCE
  - BRN30055C934CCE
  - BRN30055C934CCE
  - BRN30055C934CCE
  - Basement
  - Basement TV
  - Basement TV
  - Bedroom Apple TV
  - Blink?DHCP
  - Brother Printer
  - Colleens-iPhone
  - Colleen’s iPad
  - Colleen’s iPad
  - Davids-iPhone
  - Deb's iPhone
  - Diana’s iPhone
  - EP10
  - EP10
  - EP10
  - EP40
  - ESP_00F719
  - ESP_03FAB8
  - ESP_59C02F
  - ESP_61B7F5
  - Eloises-iPhone
  - Family Room Apple TV
  - Family Room Apple TV
  - Family Room Apple TV
  - Family Room Apple TV
  - Galaxy-Tab-A-8-0-2019
  - Garage Door
  - Great Room
  - Great Room
  - Great Room
  - Great Room
  - HS103
  - Holger's iPhone
  - Jenny’s iPhone
  - Jenny’s iPhone
  - JessicapleWatch
  - Jessicas-iPad
  - Jessie' iPhone
  - Jude's Dell Laptop
  - Jude's iPad 2020
  - Jude's iPad 2020
  - Judes-iPad-2020
  - Judes-iPhone-12
  - Jude’s iPhone 12
  - Kitchen
  - Kitchen
  - Kyra’s iPad
  - Kyra’s iPad
  - LHTU05CG1338FTN
  - LINKSYS18161
  - LINKSYS18161
  - LINKSYS18242
  - LINKSYS18242
  - LINKSYS18242
  - LINKSYS32547
  - LINKSYS37671
  - Lee’s iPhone
  - LorenaspleWatch
  - Matilda’s iPhone
  - Matilda’s iPhone
  - MatthewnsiPhone
  - Nest - Family Room
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Network Device
  - Nicholas's iPhone
  - Nicholass-iPhone
  - Oliwia’s iPhone :)
  - Rebecca Rubin’s iPhone
  - RebeccansiPhone
  - Sams actual iPad
  - Sams iPad
  - Sams iPad
  - Sams iPad
  - Sams iPad
  - Sams iPad (not)
  - Sams iPad (not)
  - Sams iPad (not)
  - Sams iPad (not)
  - Sams real iPad (not)
  - Sams real iPad (not)
  - Sams real iPad (not)
  - Sams real iPad (not)
  - Sams real iPad (not)
  - Sams real iPad (not)
  - Sarah’s iPhone
  - Sarah’s iPhone
  - Spinn
  - Susie's iPhone
  - Susie's iPhone
  - Susie's iPhone
  - SusiesApleWatch
  - Susie’s iPad
  - Susie’s iPad
  - Susie’s iPad
  - TL-SG116E
  - Tim's iPhone
  - US43-MP25CW18
  - VIZIOCastDisplay7124
  - '[TV] UN55JU6700'
  - a7cfa3e58048224c
  - a7cfa3e58048224c
  - e0c23e58-e183-454e-8bf6-417058001b90
  - e2999353-bf80-47c5-80e2-1081980a8a03
  - espressif
  - iPad
  - iPad
  - iPad
  - iPhone
  - iPhone
  - iPhone
  - iPhone
  - iPhone
  - iPhone (176)
  - iPhone (176)
  - iPhone (176)
  - iPhone (37)
  - none-3
  - none-9
  - sonos000E5883BD9C
  - sonos000E5883BD9C
  - wlan0
  - wlan0
connected_adapters: []
description: null
manufacturer: Amazon
model: AFTR
name: Basement
operating_system: Android
parent_name: null
parental_control_schedule: {}
serial: null
status: false
unique_id: 3c19d952-fd2f-487e-af46-c86000c8ff9e
friendly_name: 'Velop Mesh: Devices'

**Comment by uvjim** [comment]
Thanks. I think that's for the select. Could you grab the one for the online devices sensor?

**Comment by judetf** [comment]
Ah, sorry. Here it is:

devices:
  - name: 0e09bb49fc2974e8
    id: 2a7e34dd-7ddf-4768-aa53-498800bf97ea
    ip: 10.238.1.218
    connection: Wireless
    guest_network: false
  - name: 10-238-1-148
    id: 653375f1-b05b-4524-9aac-94a40eff637d
    ip: 10.238.1.146
    connection: Wireless
    guest_network: false
  - name: 10-238-1-32
    id: 0196a673-0855-41a5-9d08-306536ac65b4
    ip: 10.238.1.159
    connection: Wireless
    guest_network: false
  - name: 10.238.1.117 - Sonos Sub
    id: 4f0eef42-08cd-4afb-981f-d157635e3efc
    ip: 10.238.1.117
    connection: Wired
    guest_network: false
  - name: 10.238.1.13 - Sonos Play:5
    id: 4470f3f9-673a-4bf2-8063-bc293e9f2d44
    ip: 10.238.1.13
    connection: Wired
    guest_network: false
  - name: 10.238.1.168 - Sonos Play:5
    id: 659d913a-91a2-42f8-bcc7-952411c33208
    ip: 10.238.1.168
    connection: Wired
    guest_network: false
  - name: 10.238.1.76 - Sonos Play:5
    id: 6d7497c6-d124-48a4-9afd-f446c18d463b
    ip: 10.238.1.76
    connection: Wired
    guest_network: false
  - name: AmazonDevice
    id: c63c1aab-166c-467b-9b14-e7e78428a2da
    ip: 10.238.1.27
    connection: Wireless
    guest_network: false
  - name: BRN30055C934CCE
    id: 0c8f0f09-8fa9-4883-9973-088d5e6cb1f2
    ip: 10.238.1.21
    connection: Wireless
    guest_network: false
  - name: BRN30055C934CCE
    id: 4ad60949-052e-4500-acd2-95c078ddcb2d
    ip: 10.238.1.15
    connection: Wireless
    guest_network: false
  - name: Basement TV
    id: 9e550fe4-47eb-4a21-b3c5-b314c2b00c53
    ip: 10.238.1.38
    connection: Wired
    guest_network: false
  - name: Bedroom Apple TV
    id: eb4da38d-a4e5-417b-86d7-1bab6fa78234
    ip: 10.238.1.80
    connection: Wireless
    guest_network: false
  - name: Brother Printer
    id: 7f9a96c8-86d9-4915-8381-f375fc87d8ec
    ip: 10.238.1.14
    connection: Wired
    guest_network: false
  - name: EP10
    id: 0d53481c-3ac5-4546-80ca-617dd31d243e
    ip: 10.238.1.67
    connection: Wireless
    guest_network: false
  - name: EP10
    id: 76f939b3-25b4-4dcf-be5d-022f4d072dd8
    ip: 10.238.1.192
    connection: Wireless
    guest_network: false
  - name: EP10
    id: c3ba26c2-7bea-49bd-b21b-5d0101bc22e4
    ip: 10.238.1.114
    connection: Wired
    guest_network: false
  - name: EP40
    id: ed883d8c-17e1-4dd4-bc99-0b5ac87c9bd6
    ip: 10.238.1.183
    connection: Wireless
    guest_network: false
  - name: ESP_00F719
    id: c80091e4-0f01-46ff-8670-19ecadbf382b
    ip: 10.238.1.66
    connection: Wireless
    guest_network: false
  - name: ESP_03FAB8
    id: fd076fdc-bd31-4946-a9ad-7bf67f9c9e88
    ip: 10.238.1.241
    connection: Wireless
    guest_network: false
  - name: Family Room Apple TV
    id: 306ca752-d591-4c10-a283-295e87c23fb4
    ip: 10.238.1.118
    connection: Wireless
    guest_network: false
  - name: Garage Door
    id: cbae3357-bdd9-4119-b0d2-e46a22976d7d
    ip: 10.238.1.219
    connection: Wireless
    guest_network: false
  - name: Great Room
    id: 9832dfc5-b64a-4da6-af8c-bbdc4523c0d8
    ip: 10.238.1.56
    connection: Wireless
    guest_network: false
  - name: Great Room
    id: 69f5fc37-e5e6-49af-b8ec-3921c3e4b81a
    ip: 10.238.1.123
    connection: Wired
    guest_network: false
  - name: HS103
    id: d8ba5226-0801-4b57-bd73-f0039cf475da
    ip: 10.238.1.33
    connection: Wireless
    guest_network: false
  - name: Jessie' iPhone
    id: 29d2397c-266c-4f41-ad90-5d694d837a56
    ip: 10.238.1.187
    connection: Wireless
    guest_network: false
  - name: Jude's Dell Laptop
    id: 1fee9bed-457e-4bf1-87e9-1b8b3b657a4e
    ip: 10.238.1.133
    connection: Wired
    guest_network: false
  - name: Kitchen
    id: dcd2deec-1f16-4b84-9db5-c728cf2de8d3
    ip: 10.238.1.228
    connection: Wireless
    guest_network: false
  - name: LINKSYS18161
    id: 1f42908f-2404-46c7-982d-9d8c48aa4542
    ip: 10.238.1.26
    connection: Wireless
    guest_network: false
  - name: LINKSYS18242
    id: 5dd2e2c0-fa6c-470f-8085-49835e5a7cce
    ip: 10.238.1.42
    connection: Wired
    guest_network: false
  - name: Matilda’s iPhone
    id: a9371ed2-ad62-4046-b1d8-7532bd025c18
    ip: 10.238.1.229
    connection: Wireless
    guest_network: false
  - name: Network Device
    id: 75f05915-2285-4986-866e-b8dcfc6c80f2
    ip: 10.238.1.209
    connection: Wireless
    guest_network: false
  - name: Network Device
    id: f78c28cf-dae5-4cd1-ba1f-e5fc1690d0d6
    ip: 10.238.1.245
    connection: Wireless
    guest_network: false
  - name: Sams iPad (not)
    id: 7283b220-aef6-454c-923d-9874122dba30
    ip: 10.238.1.55
    connection: Wireless
    guest_network: false
  - name: Sams iPad (not)
    id: 051d312e-ae9f-44a5-8daf-58f810424bff
    ip: 10.238.1.166
    connection: Wired
    guest_network: false
  - name: Sams iPad (not)
    id: c0fcb506-d106-4b9e-8d90-7d8c4e51563b
    ip: 10.238.1.52
    connection: Wired
    guest_network: false
  - name: Sams iPad (not)
    id: a32b122e-802d-4ff9-ab1d-2786ccdd02e7
    ip: 10.238.1.143
    connection: Wired
    guest_network: false
  - name: Sams real iPad (not)
    id: 77031677-e1a5-4b62-b113-8a90c8948c81
    ip: 10.238.1.178
    connection: Wireless
    guest_network: false
  - name: Sams real iPad (not)
    id: b859d9f6-9203-404e-a3b7-ca7718499be2
    ip: 10.238.1.188
    connection: Wireless
    guest_network: false
  - name: Sams real iPad (not)
    id: 66933c76-d933-446a-8276-4550a580dcf7
    ip: 10.238.1.189
    connection: Unknown
    guest_network: false
  - name: Sams real iPad (not)
    id: cf07b35a-763d-45d2-95a9-8fc8ad6feda6
    ip: 10.238.1.251
    connection: Wired
    guest_network: false
  - name: Sams real iPad (not)
    id: b3a4dfaf-b744-45e1-8614-8a14243eaefc
    ip: 10.238.1.122
    connection: Wireless
    guest_network: false
  - name: Sarah’s iPhone
    id: 480edefc-2e31-4951-bb39-bab0977959a6
  - name: Spinn
    id: bea3048a-9964-42db-8640-51388f2e0b53
    ip: 10.238.1.165
    connection: Wireless
    guest_network: false
  - name: Susie's iPhone
    id: 8c9f597f-0e0f-4de4-a8f4-dbad82965669
    ip: 10.238.1.161
    connection: Wireless
    guest_network: false
  - name: Susie's iPhone
    id: 2d0b399a-e118-47bc-8312-347b7807270e
    ip: 10.238.1.144
    connection: Wireless
    guest_network: false
  - name: Susie’s iPad
    id: c37ef1d1-74c1-439a-b38b-8d70906f8a6a
    ip: 10.238.1.221
    connection: Wireless
    guest_network: false
  - name: TL-SG116E
    id: 934a9b49-fcce-4d82-b97f-663d62697fcb
    ip: 10.238.1.252
    connection: Wired
    guest_network: false
  - name: VIZIOCastDisplay7124
    id: 1c56d0e5-4e7b-4a7f-a035-dbed856ad594
    ip: 10.238.1.106
    connection: Wireless
    guest_network: false
  - name: a7cfa3e58048224c
    id: edb8916d-832a-4476-8ba1-87449b593841
    ip: 10.238.1.205
    connection: Wireless
    guest_network: false
  - name: espressif
    id: 65819183-9869-4f03-bd07-7ec7756b633d
    ip: 10.238.1.138
    connection: Wireless
    guest_network: false
  - name: iPhone (176)
    id: 711e6fc9-0370-4ead-845b-880b96b3877c
    ip: 10.238.1.116
    connection: Wireless
    guest_network: false
  - name: iPhone (176)
    id: fc9c7a0e-3998-47d4-b7b6-fb37b4c45cf5
    ip: 10.238.1.28
    connection: Wireless
    guest_network: false
  - name: none-3
    id: 1c2e5b0c-ad59-4f1d-9006-08d00a6a7bc8
    ip: 10.238.1.243
    connection: Wireless
    guest_network: false
  - name: none-9
    id: 30f60c0f-a091-4958-bcee-2a92b08e983e
    ip: 10.238.1.25
    connection: Wireless
    guest_network: false
  - name: sonos000E5883BD9C
    id: 64841375-ff8d-4f5c-9449-891cebd4a312
    ip: 10.238.1.177
    connection: Wireless
    guest_network: false
  - name: sonos000E5883BD9C
    id: 6128d520-d25f-42c0-91d9-bda764c4620f
    ip: 10.238.1.54
    connection: Wireless
    guest_network: false
friendly_name: 'Velop Mesh: Online Devices'

**Comment by uvjim** [comment]
I've worked out what was causing the error now.  I'll add it to a backlog to sort out (I'm not sure if I need to solve it in the library or in the integration).

The issue was manifesting itself as an online device not having two properties available (`guest_network` and `connection`).  The changes to the card check these properties exist and if they don't they are presented as a -. This means your list now builds.

The actual issue about why the device doesn't have these properties is because it had no IPv4 address. It only has an IPv6 address and I've not been checking for that (no IPv6 here at all).

I'll close this issue and open another one for the actual problem and work on that separately. I doubt it'll make it into the next release mind you (probably the one after).

Thanks for taking the time and getting the info over to me.

**Comment by judetf** [comment]
Ok, thanks for the update. I'm not sure what device it is, but I appreciate
you sorting it out. FWIW, I don't have the guest network enabled on my
Velop network, but it doesn't sound like that's related. Just sharing in
case it's relevant though.

**Comment by uvjim** [comment]
The UI now shows correctly based on the YAML made available in #240
